### PR TITLE
Utilize gomock for metrics

### PR DESCRIPTION
**Pull request by Ardagan**
Migrate metrics client and scope to use gomock for tests.

### CODE REVIEW

File: config/dynamicconfig/development.yaml [hunk 1]
```diff
@@ -31,3 +31,5 @@ history.defaultWorkflowRetryPolicy:
 system.advancedVisibilityWritingMode:
   - value: "off"
     constraints: {}
+worker.executionsScannerEnabled:
+  - value: true
```
Comment (alexshtin) [inline config/dynamicconfig/development.yaml:35]: ?

File: service/worker/scanner/executions/scavenger.go [hunk 2]
```diff
@@ -26,6 +26,7 @@ package executions
 
 import (
 	"fmt"
+	"math/rand"
```
Comment (alexshtin) [inline service/worker/scanner/executions/scavenger.go:29]: This and next file are not from this PR, I guess.

File: common/persistence/elasticsearch/esProcessor_test.go [hunk 3]
```diff
@@ -142,7 +140,7 @@ func (s *esProcessorSuite) TestAdd() {
 	s.Equal(0, s.esProcessor.mapToAckChan.Len())
 
 	s.mockBulkProcessor.EXPECT().Add(request).Times(1)
-	s.mockMetricClient.On("StartTimer", testScope, testMetric).Return(testStopWatch).Once()
+	s.mockMetricClient.EXPECT().StartTimer(testScope, testMetric).Return(testStopWatch)
```
Comment (alexshtin) [inline common/persistence/elasticsearch/esProcessor_test.go:143]: `Times(1)` seems to be default in `gomock`. Can you remove it from everywhere with next PR?

File: host/testcluster.go [hunk 4]
```diff
@@ -101,6 +101,8 @@ type (
 		EnableIndexer    bool
 		EnableReplicator bool
 	}
+
+	NoopMetricsClient struct{}
```
Comment (alexshtin) [inline host/testcluster.go:105]: Why not to use mock here too?

Comment (Ardagan) [reply host/testcluster.go:105]: I don't think we should use mocks in integration tests. Also, specifying all methods to be called inside mock is same amount of code as defining noop client and is more misleading. 
If you think using mock is still better, I can refactor this code.

File: common/persistence/elasticsearch/esProcessor_test.go (hunk 3 above)
Comment (Ardagan) [reply common/persistence/elasticsearch/esProcessor_test.go:143]: Yes. But I support this to be a separate PR.

File: host/testcluster.go (hunk 4 above)
Comment (alexshtin) [reply host/testcluster.go:105]: This is essentially mock too. I am ok to keep it. Please just move it to separate file.

**Review by alexshtin — APPROVED** [review]

**Pull request merged**